apps/portal/app/routes/readonly+/claim+/$id.tsx

```diff
@@ -131,67 +158,89 @@ export interface ReadOnlyClaimDetailsLoaderData {
 }
 
 export default function ReadOnlyClaimDetails() {
-  const { claim } = useLiveLoader<{
+  const { claim, initialParams } = useLiveLoader<{
     claim: ClaimPresenter
+    initialParams: {
+      id: string
+    }
   }>(['create', 'attest'])
 
+  const { data: tripleData } = useGetTripleQuery(
+    {
+      tripleId: initialParams.id,
+    },
+    {
+      queryKey: ['get-triple', { id: initialParams.id }],
+    },
+  )
+
   const leftPanel = (
     <div className="flex-col justify-start items-start gap-6 inline-flex w-full">
       <div className="flex-row flex m-auto md:hidden">
         <Claim
           size="xl"
+          maxIdentityLength={60}
           subject={{
-            variant: claim.subject?.is_user ? Identity.user : Identity.nonUser,
-            label: getAtomLabel(claim.subject as IdentityPresenter),
-            imgSrc: getAtomImage(claim.subject as IdentityPresenter),
-            id: claim.subject?.identity_id,
-            description: getAtomDescription(claim.subject as IdentityPresenter),
-            ipfsLink: getAtomIpfsLink(claim.subject as IdentityPresenter),
-            link: getAtomLink(claim.subject as IdentityPresenter),
+            variant:
+              tripleData?.triple?.subject?.type === 'Person'
+                ? Identity.user
+                : Identity.nonUser,
+            label: getAtomLabelGQL(tripleData?.triple?.subject as Atom),
+            imgSrc: getAtomImageGQL(tripleData?.triple?.subject as Atom),
+            id: tripleData?.triple?.subject?.id,
+            description: getAtomDescriptionGQL(
+              tripleData?.triple?.subject as Atom,
+            ),
+            ipfsLink: getAtomIpfsLinkGQL(tripleData?.triple?.subject as Atom),
+            link: getAtomLinkGQL(tripleData?.triple?.subject as Atom),
             linkComponent: RemixLink,
           }}
           predicate={{
-            variant: claim.predicate?.is_user
-              ? Identity.user
-              : Identity.nonUser,
-            label: getAtomLabel(claim.predicate as IdentityPresenter),
-            imgSrc: getAtomImage(claim.predicate as IdentityPresenter),
-            id: claim.predicate?.identity_id,
-            description: getAtomDescription(
-              claim.predicate as IdentityPresenter,
+            variant:
+              tripleData?.triple?.predicate?.type === 'Person'
+                ? Identity.user
+                : Identity.nonUser,
+            label: getAtomLabelGQL(tripleData?.triple?.predicate as Atom),
+            imgSrc: getAtomImageGQL(tripleData?.triple?.predicate as Atom),
+            id: tripleData?.triple?.predicate?.id,
+            description: getAtomDescriptionGQL(
+              tripleData?.triple?.predicate as Atom,
             ),
-            ipfsLink: getAtomIpfsLink(claim.predicate as IdentityPresenter),
-            link: getAtomLink(claim.predicate as IdentityPresenter),
+            ipfsLink: getAtomIpfsLinkGQL(tripleData?.triple?.predicate as Atom),
```

Comment:
Until we have our types locked in we can do this in the Identity Details route as well to avoid `any` -- even though we're still casting it gets us a bit closer!

I mentioned this in another comment, but we should think about how/where we want to keep/export all of these new GraphQL types. They leverage our codegen from the package but the types themselves as we're using them aren't part of the package. We should give this some thought now that we're getting closer!